xcdat/spatial_avg.py

```diff
@@ -324,83 +345,136 @@ def _get_weights(
-                        reg_bounds = self._set_equal_lon_bounds_to_domain(
-                            domain_bounds, reg_bounds
+                        # Ensure that domain and regional bounds are both
+                        # in the range of 0 -> 360
+                        dom_bounds = self._swap_lon_axes(dom_bounds, to=360)
+                        reg_bounds = self._swap_lon_axes(reg_bounds, to=360)
+
+                        # Some grid cells cross the prime meridian (
+                        # e.g., [-1, 1]). Check for this special case. If it
+                        # applies, return the index of that grid cell
+                        # (otherwise None) and recreate the dims with two
+                        # grid cells ([0, 1] and [359, 360]).
+                        pmb_index, dom_bounds = self._align_longitude_to_360_axis(
+                            dom_bounds
                         )
+
+                        # If the region bounds are equal, it means all
+                        # longitude values are specified (e.g., (360, 360) ==
+                        # (0, 360)). As a result, set the region lon bounds to
+                        # [0, 360].
+                        if reg_bounds[1] - reg_bounds[0] == 0:
+                            reg_bounds[0] = 0.0
+                            reg_bounds[1] = 360.0
+
+                    # Scale the domain boundaries to the regional domain
                     dom_bounds = self._scale_domain_to_region(dom_bounds, reg_bounds)
 
                 if dim == "lat":
                     # The area between two lines of latitude scales as the
                     # difference of the sine of latitude bounds. This scaling
                     # is applied here.
                     dom_bounds = np.sin(np.radians(dom_bounds))
+                    pmb_index = None
 
                 # Perform differencing to get weights and add metadata.
                 dim_wts = np.abs(dom_bounds[:, 1] - dom_bounds[:, 0])
+
+                # If pmb_index is not None it means that an extra
+                # longitude bound was added and the weight vector
+                # is too long. This extra weight should be added to the
+                # grid cell that crossed the prime meridian (and then
+                # removed from the dim_wts vector).
+                if pmb_index is not None:
+                    dim_wts[pmb_index] = dim_wts[pmb_index] + dim_wts[-1]
+                    dim_wts = dim_wts[0:-1]
+
+                # ensure attributes remain on dim_wts
                 dim_wts.attrs = dom_bounds.attrs
                 dim_wts.name = dim + "_wts"
                 axis_weights[dim] = dim_wts
 
         weights = self._combine_weights(axis, axis_weights)
         return weights
 
-    def _force_lon_linearity(
-        self, domain_bounds: xr.DataArray, region_bounds: np.ndarray
-    ) -> Tuple[xr.DataArray, np.ndarray]:
+    def _align_longitude_to_360_axis(self, domain_bounds: xr.DataArray):
```

Comment:
Replace `_force_lon_linearity` with `_align_longitude_to_360_axis`. It essentially ensures that the domain bounds neatly span from 0 to 360 degrees. If there is a grid cell that crosses the prime meridian (e.g., [-1, 1]) it breaks this into to different grid cells spanning [0, 1] and [359, 360]. In this case, the original grid cell index is returned with the function. The domain_bounds shape goes from [nlon, 2] to [nlon+1, 2]. When these weights are calculated (e.g., 1 - 0 = 1 and 360 - 359 = 1) they are summed and both weights are included with the grid cell that was indexed (so that the weight vector is length nlon).